### PR TITLE
Fix positive test abort condition for ENS key provision (EXPOSUREAPP-6879)

### PR DESCRIPTION
**Pull request by d4rken**
We want to abort if there is a positive test, and not just when we are allowed to submit (which is transient).

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DownloadDiagnosisKeysTask.kt [hunk 1]
```diff
@@ -114,10 +114,10 @@ class DownloadDiagnosisKeysTask @Inject constructor(
             // remember version code of this execution for next time
             settings.updateLastVersionCodeToCurrent()
 
-            val isAllowedToSubmitKeys = coronaTestRepository.coronaTests.first().any { it.isSubmissionAllowed }
-            if (isAllowedToSubmitKeys) {
-                Timber.tag(TAG).i("task aborted, positive test result")
-                return object : Task.Result {}
+            val isPositive = coronaTestRepository.coronaTests.first().any { it.isPositive }
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/DownloadDiagnosisKeysTask.kt:117]: Naming suggestion `isPositive` => `hasPositiveTest` ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/execution/PresenceTracingWarningTask.kt [hunk 2]
```diff
@@ -89,7 +90,13 @@ class PresenceTracingWarningTask @Inject constructor(
 
             presenceTracingRiskRepository.reportCalculation(successful = true)
 
-            return Result(calculatedAt = nowUTC)
+            return Result()
+        }
+
+        val isPositive = coronaTestRepository.coronaTests.first().any { it.isPositive }
```
Comment (LukasLechnerDev) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/execution/PresenceTracingWarningTask.kt:96]: Naming suggestion `isPositive` => `hasPositiveTest` ?

**Review by LukasLechnerDev — APPROVED** [review]
LGTM. 

One renaming suggestion.

**Review by mtwalli — APPROVED** [review]
lgtm

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/execution/PresenceTracingWarningTask.kt (hunk 2 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/presencetracing/risk/execution/PresenceTracingWarningTask.kt:96]: Hm, feels wrong. We are asking the `CoronaTest` if it has a positive test? The `CoronaTest` IS the positive test :smile:

**Pull request merged**